app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -711,7 +711,7 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
+            if (typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount) {
```

Comment:
Two things about this:'
1. Always use strict `!==` comparison rather than loose `!=` comparison, since the coercion rules of the latter can be surprising.
2. Under what circumstances would `result` not exist? Can we eliminate that circumstance?